### ISSUE TITLE
New feature : Add join during game option

### ISSUE THREAD

**Issue by xaviermarchal** [pokerth/pokerth#381]
As a user, it would be nice to add an option (or set it by default), to enable the possibility to join during a tournament already launched. 
Another option (Nice to have) would be to set a limited time (like 30 min) after starting the game where users can join.

It would unblock two scenarios:
1. One user is late when the tournament is starting, he can join later.
2. If one user is eliminated, he can re-join (one more cave).

**Comment by q4z1** [comment]
Hi,

as more and more people do request this feature (e.g. rebuy) it might be really an idea to implement for custom games.

But it will take time. Dev Team is really small ;-)

Regards.

**Comment by QuentinFarizon** [comment]
@q4z1 Would you be open to a PR ? I could try if you give me a starting point 
Thanks !

**Comment by q4z1** [comment]
> @q4z1 Would you be open to a PR ? I could try if you give me a starting point
> Thanks !

What do you mean by PR?

We can have a talk on our discord channel maybe:

https://discord.gg/d2Sm3wq

Regards.

**Comment by QuentinFarizon** [comment]
Pull request / merge request

Le lun. 30 mars 2020 à 22:40, Kai Philipp <notifications@github.com> a
écrit :

**Comment by q4z1** [comment]
Ahh ... best way to start is to fork the pokerth git repository.

Then switch to stable branch and start building. Details are in the Wiki:

https://github.com/pokerth/pokerth/wiki

Good luck ;-)

**Comment by QuentinFarizon** [comment]
Yes sure, I meant if you could give me a pointer on some classes / files that would need to be changed, as a starting point ?

**Comment by q4z1** [comment]
Now I get you :D

First of all - albmed and me did not develop the code. We just do bugfixes and minor extensions.

For the server:
`src/net/common/servergame.cpp`
`src/net/common/servergamestate.cpp`

I would start here as the servergame itself is managed here.

`src/net/common/serverlobbythread.cpp`

Maybe worth a look too.

---

For the client:

I never dived into the client/GUI so far ... but:

`src/gui/`
`src/engine/`

Here the client/GUI actions take place.
----

You have to figure out how to get a player back into the game - the server keeps the players until end of game - so that should be possible. I'd start with serverside-only coding, bringing a player automatically back when he's on 0 maybe. That should be easy to test/simulate.

For the rebuy-action the network protocol hast to be extended by extra "commands".
These have to be implemented both on server and client side.

I hope that helps.

Regards.

**Comment by janekg89** [comment]
I would like the option of multiple rebuys this would also enable cash-style games.  #77.
Many thanks for your efforts

**Comment by Mantilla2101** [comment]
The option to set rebuys for a adjustable limited time would be such a important feature! Please add it. Many thanks!

**Comment by skipple** [comment]
Just commenting to bump this request. Would love to see this feature added.

**Comment by Pokerasmo** [comment]
I would also love this feature. I am willing to donate 100Euros for it.

**Comment by q4z1** [comment]
We slowly started to work on it. Really not sure, how long it takes.

**Comment by lordmabl** [comment]
Would be great if you could keep us up to track about the progress

**Comment by q4z1** [comment]
@albmed started with it. He already found some concept issues which are not that easy to solve. We cannot say better than: it will be longer.

**Comment by NitramTsuj** [comment]
Hey There.
Are there any new infos?

**Comment by q4z1** [comment]
No more plans on that.
